pkg/testing/ogc/provisioner.go

```diff
@@ -0,0 +1,338 @@
+	}
+	var output bytes.Buffer
+	p.logger.Logf("Pulling latest ogc image")
+	proc, err := process.Start("docker", process.WithContext(ctx), process.WithArgs(args), process.WithCmdOptions(runner.AttachOut(&output), runner.AttachErr(&output)))
+	if err != nil {
+		return fmt.Errorf("failed to run docker ogcPull: %w", err)
+	}
+	ps := <-proc.Wait()
+	if ps.ExitCode() != 0 {
+		// print the output so its clear what went wrong
+		fmt.Fprintf(os.Stdout, "%s\n", output.Bytes())
+		return fmt.Errorf("failed to run ogc pull: docker run exited with code: %d", ps.ExitCode())
+	}
+	return nil
+}
+
+// ogcImport imports all the required batches into OGC.
+func (p *provisioner) ogcImport(ctx context.Context, batches []runner.OSBatch) error {
+	var layouts []Layout
+	for _, ob := range batches {
+		layouts = append(layouts, osBatchToOGC(ob))
+	}
+	layoutData, err := yaml.Marshal(struct {
+		Layouts []Layout `yaml:"layouts"`
+	}{
+		Layouts: layouts,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to marshal layouts YAML: %w", err)
+	}
+
+	var output bytes.Buffer
+	p.logger.Logf("Import layouts into ogc")
+	proc, err := p.ogcRun(ctx, []string{"layout", "import"}, true, process.WithCmdOptions(runner.AttachOut(&output), runner.AttachErr(&output)))
+	if err != nil {
+		return fmt.Errorf("failed to run ogc import: %w", err)
+	}
+	_, err = proc.Stdin.Write(layoutData)
+	if err != nil {
+		_ = proc.Stdin.Close()
+		_ = proc.Kill()
+		<-proc.Wait()
+		// print the output so its clear what went wrong
+		fmt.Fprintf(os.Stdout, "%s\n", output.Bytes())
+		return fmt.Errorf("failed to write layouts to stdin: %w", err)
+	}
+	_ = proc.Stdin.Close()
+	ps := <-proc.Wait()
+	if ps.ExitCode() != 0 {
+		// print the output so its clear what went wrong
+		fmt.Fprintf(os.Stdout, "%s\n", output.Bytes())
+		return fmt.Errorf("failed to run ogc import: docker run exited with code: %d", ps.ExitCode())
+	}
+	return nil
+}
+
+// ogcUp brings up all the instances.
+func (p *provisioner) ogcUp(ctx context.Context) ([]byte, error) {
+	p.logger.Logf("Bring up instances through ogc")
+	var output bytes.Buffer
+	proc, err := p.ogcRun(ctx, []string{"up", LayoutIntegrationTag}, false, process.WithCmdOptions(runner.AttachOut(&output), runner.AttachErr(&output)))
+	if err != nil {
+		return nil, fmt.Errorf("failed to run ogc up: %w", err)
+	}
+	ps := <-proc.Wait()
+	if ps.ExitCode() != 0 {
+		// print the output so its clear what went wrong
+		fmt.Fprintf(os.Stdout, "%s\n", output.Bytes())
+		return nil, fmt.Errorf("failed to run ogc up: docker run exited with code: %d", ps.ExitCode())
+	}
+	return output.Bytes(), nil
+}
+
+// ogcDown brings down all the instances.
+func (p *provisioner) ogcDown(ctx context.Context) error {
+	p.logger.Logf("Bring down instances through ogc")
+	var output bytes.Buffer
+	proc, err := p.ogcRun(ctx, []string{"down", LayoutIntegrationTag}, false, process.WithCmdOptions(runner.AttachOut(&output), runner.AttachErr(&output)))
```

Comment:
Rather than LayoutIntegrationTag, you can use `ogc down` or `ogc down -q 'instance_name=blah'` for individual ones

Reply:
When I test out the `:bleed` tag I will make this change.